src/PrimeTower.cpp

```diff
@@ -102,8 +102,9 @@ void PrimeTower::generateGroundpoly()
 
 void PrimeTower::generatePaths(const SliceDataStorage& storage)
 {
-    would_have_actual_tower
-        = storage.max_print_height_second_to_last_extruder >= 0; // Maybe it turns out that we don't need a prime tower after all because there are no layer switches.
+    const int raft_total_extra_layers = Raft::getTotalExtraLayers();
+    would_have_actual_tower = storage.max_print_height_second_to_last_extruder
+                           >= -raft_total_extra_layers + 1; // Maybe it turns out that we don't need a prime tower after all because there are no layer switches.
```

Comment:
```suggestion
                           >= -raft_total_extra_layers; // Maybe it turns out that we don't need a prime tower after all because there are no layer switches.
```